### ISSUE TITLE
Allow user to choose terminal for Detached Execution

### ISSUE THREAD

**Issue by cs-mshah** [cpeditor/cpeditor#474]
**Describe the bug**

On pressing the specified key binding for run detached mode, i.e ctrl+alt+D, the editor minimises instead of opening the terminal. I think this should be fixed.

**Enhancement**

For ubuntu users, the xterm doesn't come pre-installed. We get the genome-terminal. So a support for that could also be added instead of the only available xterm. One would naturally use the terminal of his choice to have the its saved configured features.

**Environment**

- App version: 6.4.7
- Git commit hash: e352209
- Build time: Jun 19 2020 05:17:51
- OS: Linux

**Comment by ouuan** [comment]
> On pressing the specified key binding for run detached mode, i.e ctrl+alt+D, the editor minimises instead of opening the terminal.

I guess that's a shortcut on your system/windows manager, and not related to CP Editor. You can set a custom shortcut after #214 is finished.

> For ubuntu users, the xterm doesn't come pre-installed. We get the genome-terminal. So a support for that could also be added instead of the only available xterm. One would naturally use the terminal of his choice to have the its saved configured features.

We've considered supporting some popular terminals, but we found that it's hard to determine which terminal the user is using. So we planned to make the detached run command customizable, but that's also not easy, because it's not enough to provide some place holders like `${run-command}`, we also have to make it able to escape quotes. Another choice is to only support xterm, konsole and gnome-terminal, and it can hopefully cover most Linux users.

**Comment by coder3101** [comment]
> we planned to make the detached run command customizable, but that's also not easy, because it's not enough to provide some place holders like `${run-command}`, we also have to make it able to escape quotes.

Agreed!

We can provide a setting for Linux users, to choose from `xterm`, `konsole` or `gnome-terminal`, It is difficult to find the DE of Linux. Defaulting to `xterm` 

Updating triage to enhancement.

**Comment by coder3101** [comment]
@cs-mshah You should have opened 2 issues one for enhancement and another for the bug. In order to credit you for the enhancement idea, I have changed it into Enhancement Issue. If you think your issue of "Detached execution not working" exists then please open another issue.

**Comment by coder3101** [comment]
Here is a better approach for a script that runs detached.

xterm: `./a.out ; echo '\nExecution Completed\nPress enter to exit' ; read`
konsole: `<awaited>`
gnome-terminal: `<awaited>`

Why not read -n 1 -p "...."?
`read` is a bash function and does not have full options in other shells. For instance I use `zsh` as default shell and `read` flags are limited. So, We should use less and less shell specific flags. `read and echo` are generally supported in all shells.
